src/shared_types.rs

```diff
@@ -0,0 +1,283 @@
+use serde::de;
+use serde::de::Deserializer;
+use serde::ser;
+use serde::ser::{SerializeStruct, Serializer};
+use serde::{Deserialize, Serialize};
+
+#[cfg(feature = "druid")]
+use druid::Data;
+
+/// Identifiers are optional attributes of several objects in the UFO.
+/// These identifiers are required to be unique within certain contexts
+/// as defined on a per object basis throughout this specification.
+/// Identifiers are specified as a string between one and 100 characters long.
+/// All characters must be in the printable ASCII range, 0x20 to 0x7E.
+#[derive(Debug, Clone, PartialEq)]
+pub struct Identifier(pub(crate) String);
+
+impl Identifier {
+    fn is_valid(&self) -> bool {
+        if self.0.len() > 100 {
+            return false;
+        }
+        for c in self.0.chars() {
+            if !(0x20..=0x7F).contains(&(c as u8)) {
```

Comment:
`..=` is an *inclusive* range, that is it includes the last item; `0..=2` is the values `[0, 1, 2]`. An *exclusive* range is written `0..2`. 

This is a little rust thing to keep in mind, I've definitely gotten it confused in the past.

Reply:
Yes, I mean to include `0x7F` here :)

Reply:
Oops, my mistake. I wouldn't normally consider 0x7F a printable character, but whatever the spec says.

Reply:
Ooooh! Actually you are right, spec says 0x20 to 0x7E. 😯

Reply:
😅 💦